pkg/asset/installconfig/kubevirt/client.go

```diff
@@ -0,0 +1,255 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kubevirt
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	nadv1 "github.com/k8snetworkplumbingwg/network-attachment-definition-client/pkg/apis/k8s.cni.cncf.io/v1"
+	corev1 "k8s.io/api/core/v1"
+	storagev1 "k8s.io/api/storage/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	meta_v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	unstructured "k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+	kubevirtapiv1 "kubevirt.io/client-go/api/v1"
+	cdiv1 "kubevirt.io/containerized-data-importer/pkg/apis/core/v1alpha1"
+)
+
+//go:generate mockgen -source=./client.go -destination=./mock/client_generated.go -package=mock
+
+// ClientBuilderFuncType is function type for building infra-cluster clients
+type ClientBuilderFuncType func() (Client, error)
+
+// Client is a wrapper object for actual infra-cluster clients: kubernetes and the kubevirt
+type Client interface {
+	GetVirtualMachine(namespace string, name string) (*kubevirtapiv1.VirtualMachine, error)
+	ListVirtualMachine(namespace string) (*kubevirtapiv1.VirtualMachineList, error)
+	DeleteVirtualMachine(namespace string, name string) error
+	GetDataVolume(namespace string, name string) (*cdiv1.DataVolume, error)
+	ListDataVolume(namespace string) (*cdiv1.DataVolumeList, error)
+	DeleteDataVolume(namespace string, name string) error
+	GetSecret(namespace string, name string) (*corev1.Secret, error)
+	ListSecret(namespace string) (*corev1.SecretList, error)
+	DeleteSecret(namespace string, name string) error
+	GetStorageClass(ctx context.Context, name string) (*storagev1.StorageClass, error)
+	GetNetworkAttachmentDefinition(ctx context.Context, name string, namespace string) (*unstructured.Unstructured, error)
+}
+
+type client struct {
+	kubernetesClient *kubernetes.Clientset
+	dynamicClient    dynamic.Interface
+}
+
+// NewClient creates our client wrapper object for the actual kubeVirt and kubernetes clients we use.
+func NewClient() (Client, error) {
+	loadingRules := clientcmd.NewDefaultClientConfigLoadingRules()
+	// if you want to change the loading rules (which files in which order), you can do so here
```

Comment:
those comments are from the example, I'm not sure we need those here.

Reply:
Removed comments